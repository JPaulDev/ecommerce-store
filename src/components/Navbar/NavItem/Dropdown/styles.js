import styled from 'styled-components';

export const Container = styled.div`
  background-color: white;
  box-shadow: 0 5px 5px 0 rgb(0 0 0 / 20%);
  display: flex;
  left: 50%;
  max-width: 1100px;
  position: absolute;
  transform: translateX(-50%);
  width: 96%;
`;

export const ColumnContainer = styled.div`
  flex: 1;
  padding: 30px;
`;

export const Heading = styled.h4`
  border-bottom: 1px solid var(--color-orange-1);
  color: var(--color-orange-1);
  font-size: 1.03rem;
  margin-bottom: 10px;
  padding-bottom: 7px;
`;

export const List = styled.ul`
  list-style: none;
`;

export const ListItem = styled.li`
  padding-bottom: 7px;
`;

export const Link = styled.a`
  color: var(--color-grey-2);
  text-decoration: none;

  &:hover {
    color: var(--color-grey-5);
    text-decoration: underline;
  }
`;
