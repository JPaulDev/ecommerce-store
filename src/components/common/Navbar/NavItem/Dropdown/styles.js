import styled from 'styled-components';

export const Container = styled.div`
  background-color: white;
  box-shadow: var(--box-shadow-xl);
  display: flex;
  left: 50%;
  max-width: 1100px;
  position: absolute;
  transform: translateX(-50%);
  width: 96%;
`;

export const ColumnContainer = styled.div`
  flex: 1;
  padding: 30px;
`;

export const Heading = styled.h4`
  border-bottom: 1px solid var(--color-orange-1);
  color: var(--color-orange-1);
  margin-bottom: 10px;
  padding-bottom: 7px;
`;

export const List = styled.ul`
  list-style: none;

  li {
    padding-bottom: 7px;
  }
`;

export const Link = styled.a`
  color: var(--color-grey-2);
  text-decoration: none;

  &:hover {
    color: var(--color-grey-5);
    text-decoration: underline;
  }
`;

export const AdvertsContainer = styled.div`
  background-color: var(--color-grey-15);
  display: flex;
  flex-direction: column;
  padding: 25px;
  row-gap: 15px;

  img {
    border-radius: var(--border-radius-xl);
    box-shadow: var(--box-shadow-lg);
  }

  @media screen and (max-width: 924px) {
    display: none;
  }
`;
