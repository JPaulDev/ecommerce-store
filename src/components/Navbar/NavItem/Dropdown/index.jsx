import { Fragment } from 'react';
import useMediaQuery from '../../../../hooks/useMediaQuery';
import componentsAd1 from '../../../../assets/images/dropdown-ads/components-ad1.webp';
import componentsAd2 from '../../../../assets/images/dropdown-ads/components-ad2.webp';
import componentsAd3 from '../../../../assets/images/dropdown-ads/components-ad3.webp';
import accessoriesAd1 from '../../../../assets/images/dropdown-ads/accessories-ad1.webp';
import accessoriesAd2 from '../../../../assets/images/dropdown-ads/accessories-ad2.webp';
import * as Styled from './styles';

const dropdowns = {
  components: [
    [
      {
        category: 'Processors',
        links: [
          { text: 'AMD', url: '' },
          { text: 'Intel', url: '' },
        ],
      },
      {
        category: 'Memory',
        links: [
          { text: 'DDR5', url: '' },
          { text: 'DDR4', url: '' },
          { text: 'DDR3', url: '' },
        ],
      },
      {
        category: 'Motherboards',
        links: [
          { text: 'AMD Motherboards', url: '' },
          { text: 'Intel Motherboards', url: '' },
        ],
      },
    ],
    [
      {
        category: 'Graphics Cards',
        links: [
          { text: 'NVIDIA Graphics Cards', url: '' },
          { text: 'AMD Graphics Cards', url: '' },
        ],
      },
      {
        category: 'Power Supplies',
        links: [
          { text: 'All PSUs', url: '' },
          { text: 'Fully Wired', url: '' },
          { text: 'Fully Modular', url: '' },
        ],
      },
    ],
    [
      {
        category: 'Cases',
        links: [
          { text: 'Full Tower', url: '' },
          { text: 'Midi Tower', url: '' },
          { text: 'ITX', url: '' },
        ],
      },
      {
        category: 'Storage',
        links: [{ text: 'Solid State Drives', url: '' }],
      },
    ],
  ],
  accessories: [
    [
      {
        category: 'Keyboards',
        links: [
          { text: 'Wired Keyboards', url: '' },
          { text: 'Wireless Keyboards', url: '' },
          { text: 'Gaming Keyboards', url: '' },
        ],
      },
      {
        category: 'Mice',
        links: [
          { text: 'Wired Mice', url: '' },
          { text: 'Wireless Mice', url: '' },
        ],
      },
    ],
    [
      {
        category: 'Audio',
        links: [
          { text: 'Microphones', url: '' },
          { text: 'Speakers', url: '' },
        ],
      },
    ],
    [
      {
        category: 'Monitors',
        links: [
          { text: 'Gaming Monitors', url: '' },
          { text: 'Ultrawide Monitors', url: '' },
          { text: 'OLED Monitors', url: '' },
        ],
      },
    ],
  ],
};

const adverts = {
  components: [componentsAd1, componentsAd2, componentsAd3],
  accessories: [accessoriesAd1, accessoriesAd2],
};

export default function Dropdown({ menu }) {
  const isMatch = useMediaQuery('(min-width: 925px)');
  const openMenu = menu.toLowerCase().trim();
  const { [openMenu]: dropdown } = dropdowns;

  return (
    <Styled.Container data-testid="dropdown-menu">
      {dropdown.map((column, index) => (
        <Styled.ColumnContainer key={index}>
          {column.map((item) => (
            <Fragment key={item.category}>
              <Styled.Heading>{item.category}</Styled.Heading>
              <Styled.List>
                {item.links.map((link) => (
                  <Styled.ListItem key={link.text}>
                    <Styled.Link href={link.url}>{link.text}</Styled.Link>
                  </Styled.ListItem>
                ))}
              </Styled.List>
            </Fragment>
          ))}
        </Styled.ColumnContainer>
      ))}
      {isMatch && (
        <Styled.AdvertsContainer>
          {adverts[openMenu].map((image, index) => (
            <Styled.Image key={index} src={image} alt="" />
          ))}
        </Styled.AdvertsContainer>
      )}
    </Styled.Container>
  );
}
