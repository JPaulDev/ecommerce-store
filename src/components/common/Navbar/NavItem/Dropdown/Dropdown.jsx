import Link from 'next/link';
import Image from 'next/future/image';
import { Fragment } from 'react';
import componentsAd1 from '../../../../../../public/images/dropdown-ads/components-ad1.webp';
import componentsAd2 from '../../../../../../public/images/dropdown-ads/components-ad2.webp';
import componentsAd3 from '../../../../../../public/images/dropdown-ads/components-ad3.webp';
import accessoriesAd1 from '../../../../../../public/images/dropdown-ads/accessories-ad1.webp';
import accessoriesAd2 from '../../../../../../public/images/dropdown-ads/accessories-ad2.webp';
import * as Styled from './styles';

const dropdowns = {
  components: [
    [
      {
        category: 'Processors',
        links: [
          { text: 'AMD', href: '/shop/components/processors/amd' },
          { text: 'Intel', href: '/shop/components/processors/intel' },
        ],
      },
      {
        category: 'Memory',
        links: [
          { text: 'DDR5', href: '/shop/components/memory/ddr5' },
          { text: 'DDR4', href: '/shop/components/memory/ddr4' },
        ],
      },
      {
        category: 'Motherboards',
        links: [
          {
            text: 'AMD Motherboards',
            href: '/shop/components/motherboards/amd',
          },
          {
            text: 'Intel Motherboards',
            href: '/shop/components/motherboards/intel',
          },
        ],
      },
    ],
    [
      {
        category: 'Graphics Cards',
        links: [
          {
            text: 'NVIDIA Graphics Cards',
            href: '/shop/components/graphics-cards/nvidia',
          },
          {
            text: 'AMD Graphics Cards',
            href: '/shop/components/graphics-cards/amd',
          },
        ],
      },
      {
        category: 'Power Supplies',
        links: [
          {
            text: 'Fully Wired',
            href: '/shop/components/power-supplies/wired',
          },
          {
            text: 'Fully Modular',
            href: '/shop/components/power-supplies/modular',
          },
        ],
      },
    ],
    [
      {
        category: 'Cases',
        links: [
          { text: 'Full Tower', href: '/shop/components/cases/full-tower' },
          { text: 'Midi Tower', href: '/shop/components/cases/midi-tower' },
          { text: 'ITX', href: '/shop/components/cases/itx' },
        ],
      },
      {
        category: 'Storage',
        links: [
          {
            text: 'Solid State Drives',
            href: '/shop/components/storage/solid-state-drives',
          },
        ],
      },
    ],
  ],
  accessories: [
    [
      {
        category: 'Keyboards',
        links: [
          { text: 'Wired Keyboards', href: '' },
          { text: 'Wireless Keyboards', href: '' },
          { text: 'Gaming Keyboards', href: '' },
        ],
      },
      {
        category: 'Mice',
        links: [
          { text: 'Wired Mice', href: '' },
          { text: 'Wireless Mice', href: '' },
        ],
      },
    ],
    [
      {
        category: 'Audio',
        links: [
          { text: 'Microphones', href: '' },
          { text: 'Speakers', href: '' },
        ],
      },
    ],
    [
      {
        category: 'Monitors',
        links: [
          { text: 'Gaming Monitors', href: '' },
          { text: 'Ultrawide Monitors', href: '' },
          { text: 'OLED Monitors', href: '' },
        ],
      },
    ],
  ],
};

const adverts = {
  components: [componentsAd1, componentsAd2, componentsAd3],
  accessories: [accessoriesAd1, accessoriesAd2],
};

export default function Dropdown({ menu = '' }) {
  const openMenu = menu.toLowerCase().trim();

  return (
    <Styled.Container data-testid="dropdown-menu" id={menu}>
      {dropdowns[openMenu]?.map((column, index) => (
        <Styled.ColumnContainer key={index}>
          {column.map((item) => (
            <Fragment key={item.category}>
              <Styled.Heading>{item.category}</Styled.Heading>
              <Styled.List>
                {item.links.map((link) => (
                  <li key={link.text}>
                    <Link href={link.href} passHref>
                      <Styled.Link>{link.text}</Styled.Link>
                    </Link>
                  </li>
                ))}
              </Styled.List>
            </Fragment>
          ))}
        </Styled.ColumnContainer>
      ))}
      <Styled.AdvertsContainer>
        {adverts[openMenu]?.map((image, index) => (
          <Image key={index} src={image} alt="" />
        ))}
      </Styled.AdvertsContainer>
    </Styled.Container>
  );
}
