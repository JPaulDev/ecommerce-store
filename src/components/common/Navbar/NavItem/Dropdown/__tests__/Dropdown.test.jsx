import renderer from 'react-test-renderer';
import Dropdown from '../index';

describe('Dropdown component', () => {
  it('renders the components menu with all links/advertisements present', () => {
    const tree = renderer.create(<Dropdown menu="Components" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the accessories menu with all links/advertisements present', () => {
    const tree = renderer.create(<Dropdown menu="Accessories" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
