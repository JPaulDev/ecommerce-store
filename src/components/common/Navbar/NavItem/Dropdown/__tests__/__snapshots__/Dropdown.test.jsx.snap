// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown component renders the accessories menu with all links/advertisements present 1`] = `
<div
  className="sc-761e3646-0 dezeyQ"
  data-testid="dropdown-menu"
  id="Accessories"
>
  <div
    className="sc-761e3646-1 dnHZgB"
  >
    <h4
      className="sc-761e3646-2 jmPIqe"
    >
      Keyboards
    </h4>
    <ul
      className="sc-761e3646-3 qKKzw"
    >
      <li>
        <a
          className="sc-761e3646-4 jELmN"
          href=""
        >
          Wired Keyboards
        </a>
      </li>
      <li>
        <a
          className="sc-761e3646-4 jELmN"
          href=""
        >
          Wireless Keyboards
        </a>
      </li>
      <li>
        <a
          className="sc-761e3646-4 jELmN"
          href=""
        >
          Gaming Keyboards
        </a>
      </li>
    </ul>
    <h4
      className="sc-761e3646-2 jmPIqe"
    >
      Mice
    </h4>
    <ul
      className="sc-761e3646-3 qKKzw"
    >
      <li>
        <a
          className="sc-761e3646-4 jELmN"
          href=""
        >
          Wired Mice
        </a>
      </li>
      <li>
        <a
          className="sc-761e3646-4 jELmN"
          href=""
        >
          Wireless Mice
        </a>
      </li>
    </ul>
  </div>
  <div
    className="sc-761e3646-1 dnHZgB"
  >
    <h4
      className="sc-761e3646-2 jmPIqe"
    >
      Audio
    </h4>
    <ul
      className="sc-761e3646-3 qKKzw"
    >
      <li>
        <a
          className="sc-761e3646-4 jELmN"
          href=""
        >
          Microphones
        </a>
      </li>
      <li>
        <a
          className="sc-761e3646-4 jELmN"
          href=""
        >
          Speakers
        </a>
      </li>
    </ul>
  </div>
  <div
    className="sc-761e3646-1 dnHZgB"
  >
    <h4
      className="sc-761e3646-2 jmPIqe"
    >
      Monitors
    </h4>
    <ul
      className="sc-761e3646-3 qKKzw"
    >
      <li>
        <a
          className="sc-761e3646-4 jELmN"
          href=""
        >
          Gaming Monitors
        </a>
      </li>
      <li>
        <a
          className="sc-761e3646-4 jELmN"
          href=""
        >
          Ultrawide Monitors
        </a>
      </li>
      <li>
        <a
          className="sc-761e3646-4 jELmN"
          href=""
        >
          OLED Monitors
        </a>
      </li>
    </ul>
  </div>
  <div
    className="sc-761e3646-5 glabts"
  >
    <img
      alt=""
      data-nimg="future"
      decoding="async"
      height={24}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
      srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
      style={
        {
          "color": "transparent",
        }
      }
      width={24}
    />
    <img
      alt=""
      data-nimg="future"
      decoding="async"
      height={24}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
      srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
      style={
        {
          "color": "transparent",
        }
      }
      width={24}
    />
  </div>
</div>
`;

exports[`Dropdown component renders the components menu with all links/advertisements present 1`] = `
<div
  className="sc-761e3646-0 dezeyQ"
  data-testid="dropdown-menu"
  id="Components"
>
  <div
    className="sc-761e3646-1 dnHZgB"
  >
    <h4
      className="sc-761e3646-2 jmPIqe"
    >
      Processors
    </h4>
    <ul
      className="sc-761e3646-3 qKKzw"
    >
      <li>
        <a
          className="sc-761e3646-4 jELmN"
          href="/shop/components/processors-amd"
        >
          AMD
        </a>
      </li>
      <li>
        <a
          className="sc-761e3646-4 jELmN"
          href="/shop/components/processors-intel"
        >
          Intel
        </a>
      </li>
    </ul>
    <h4
      className="sc-761e3646-2 jmPIqe"
    >
      Memory
    </h4>
    <ul
      className="sc-761e3646-3 qKKzw"
    >
      <li>
        <a
          className="sc-761e3646-4 jELmN"
          href="/shop/components/memory-ddr5"
        >
          DDR5
        </a>
      </li>
      <li>
        <a
          className="sc-761e3646-4 jELmN"
          href="/shop/components/memory-ddr4"
        >
          DDR4
        </a>
      </li>
    </ul>
    <h4
      className="sc-761e3646-2 jmPIqe"
    >
      Motherboards
    </h4>
    <ul
      className="sc-761e3646-3 qKKzw"
    >
      <li>
        <a
          className="sc-761e3646-4 jELmN"
          href="/shop/components/motherboards-amd"
        >
          AMD Motherboards
        </a>
      </li>
      <li>
        <a
          className="sc-761e3646-4 jELmN"
          href="/shop/components/motherboards-intel"
        >
          Intel Motherboards
        </a>
      </li>
    </ul>
  </div>
  <div
    className="sc-761e3646-1 dnHZgB"
  >
    <h4
      className="sc-761e3646-2 jmPIqe"
    >
      Graphics Cards
    </h4>
    <ul
      className="sc-761e3646-3 qKKzw"
    >
      <li>
        <a
          className="sc-761e3646-4 jELmN"
          href="/shop/components/graphics-cards-nvidia"
        >
          NVIDIA Graphics Cards
        </a>
      </li>
      <li>
        <a
          className="sc-761e3646-4 jELmN"
          href="/shop/components/graphics-cards-amd"
        >
          AMD Graphics Cards
        </a>
      </li>
    </ul>
    <h4
      className="sc-761e3646-2 jmPIqe"
    >
      Power Supplies
    </h4>
    <ul
      className="sc-761e3646-3 qKKzw"
    >
      <li>
        <a
          className="sc-761e3646-4 jELmN"
          href="/shop/components/power-supplies-wired"
        >
          Fully Wired
        </a>
      </li>
      <li>
        <a
          className="sc-761e3646-4 jELmN"
          href="/shop/components/power-supplies-modular"
        >
          Fully Modular
        </a>
      </li>
    </ul>
  </div>
  <div
    className="sc-761e3646-1 dnHZgB"
  >
    <h4
      className="sc-761e3646-2 jmPIqe"
    >
      Cases
    </h4>
    <ul
      className="sc-761e3646-3 qKKzw"
    >
      <li>
        <a
          className="sc-761e3646-4 jELmN"
          href="/shop/components/cases-full-tower"
        >
          Full Tower
        </a>
      </li>
      <li>
        <a
          className="sc-761e3646-4 jELmN"
          href="/shop/components/cases-midi-tower"
        >
          Midi Tower
        </a>
      </li>
      <li>
        <a
          className="sc-761e3646-4 jELmN"
          href="/shop/components/cases-itx"
        >
          ITX
        </a>
      </li>
    </ul>
    <h4
      className="sc-761e3646-2 jmPIqe"
    >
      Storage
    </h4>
    <ul
      className="sc-761e3646-3 qKKzw"
    >
      <li>
        <a
          className="sc-761e3646-4 jELmN"
          href="/shop/components/solid-state-drives"
        >
          Solid State Drives
        </a>
      </li>
    </ul>
  </div>
  <div
    className="sc-761e3646-5 glabts"
  >
    <img
      alt=""
      data-nimg="future"
      decoding="async"
      height={24}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
      srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
      style={
        {
          "color": "transparent",
        }
      }
      width={24}
    />
    <img
      alt=""
      data-nimg="future"
      decoding="async"
      height={24}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
      srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
      style={
        {
          "color": "transparent",
        }
      }
      width={24}
    />
    <img
      alt=""
      data-nimg="future"
      decoding="async"
      height={24}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
      srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
      style={
        {
          "color": "transparent",
        }
      }
      width={24}
    />
  </div>
</div>
`;
